src/OpenTelemetry.AutoInstrumentation/Logger/LogBuilderExtensions.cs

```diff
@@ -96,5 +116,31 @@ public static ILoggingBuilder AddOpenTelemetryLogs(this ILoggingBuilder builder)
 
         return builder;
     }
+
+    private static bool IsAutoInstrumentationStartupAssemblyConfigured()
+    {
+        // If autoinstrumentation's HostingStartupAssembly is configured, assume integration was already called.
+        _autoInstrumentationStartupAssemblyConfigured ??= IsAutoInstrumentationStartupAssemblySet();
+        return _autoInstrumentationStartupAssemblyConfigured.Value;
+    }
+
+    private static bool IsNet6()
+    {
+        var frameworkDescription = FrameworkDescription.Instance;
+        return frameworkDescription.Name == ".NET" && frameworkDescription.ProductVersion.StartsWith("6");
+    }
+
+    private static bool IsHostServiceCollection(IServiceCollection builderServices)
+    {
+        _applicationLifetimeType ??= Type.GetType("Microsoft.Extensions.Hosting.Internal.ApplicationLifetime, Microsoft.Extensions.Hosting");
+        var applicationLifetimeDescriptor = builderServices.FirstOrDefault(sd => sd.ImplementationType == _applicationLifetimeType);
+        return applicationLifetimeDescriptor != null;
+    }
+
+    private static bool IsAutoInstrumentationStartupAssemblySet()
+    {
+        var environmentVariable = Environment.GetEnvironmentVariable("ASPNETCORE_HOSTINGSTARTUPASSEMBLIES");
```

Comment:
This could throw SecurityException.

Reply:
Can you give a sample scenario when you would expect it to fail with that exception?

Reply:
Reference: https://learn.microsoft.com/en-us/dotnet/api/system.environment.getenvironmentvariable?view=net-7.0#system-environment-getenvironmentvariable(system-string)

Reply:
changed in https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/3003/commits/4553d4aa02e202baf609a8cbb2d5765c02194962